crt/glibc_stub.c

```diff
@@ -0,0 +1,224 @@
+#include <assert.h>
+#include <bits/alltypes.h>
+#include <resolv.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ucontext.h>
+
+#define weak_alias(old, new) \
+    extern __typeof(old) new __attribute__((__weak__, __alias__(#old)))
+
+__thread int errno;
+
+const char* const sys_errlist[] = {};
+const int sys_nerr = 0;
+
+void __makecontext(ucontext_t* ucp, void (*func)(void), int argc, ...)
+{
+    assert(0);
+}
+weak_alias(__makecontext, makecontext);
+
+int setcontext(const ucontext_t* ucp)
+{
+    assert(0);
+    return -1;
+}
+
+typedef struct __mbstate_t
+{
+    unsigned __opaque1, __opaque2;
+} mbstate_t;
+size_t __mbrlen(const char* s, size_t n, mbstate_t* ps)
+{
+    assert(0);
+    return 0;
+}
+weak_alias(__mbrlen, mbrlen);
+
+int __openat_2(int fd, const char* file, int oflag)
+{
+    assert(0);
+    return -1;
+}
+
+struct obstack
+{
+};
+
+void _obstack_newchunk(struct obstack* h, int length)
+{
+    assert(0);
+}
+
+int __argz_create_sep(const char* string, int delim, char** argz, size_t* len)
+{
+    assert(0);
+    return 0;
```

Comment:
https://www.man7.org/linux/man-pages/man3/argz_create_sep.3.html - return ENOMEM on error

Reply:
I'm not sure whether the implementation should return -ENOMEM or ENOMEM. @mikbras, can you chime in?